base/osutils.jl

```diff
@@ -1,20 +1,26 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
-"""
-    OS_NAME
+let UNAME = ccall(:jl_get_UNAME, Any, ())
+    # change some `uname -s` values to "friendly" names
+    # for internal usage
```

Comment:
>  one is going to write code testing against "NT."

Python isn't exactly a "nobody" (https://docs.python.org/2/library/os.html#os.name), although Python also seems to provide every other sort of name too.

> the userland OS, environment, and common names that people will want to test against and know what it means even if they've never written a shell script that called uname, do matter.

on the contrary, it is the _only_ thing that matters. this variable is used to identify the features / limitations and performance characteristics of the underlying system. anything else that can be tested for should be done as a feature test, since the userland environment is generally meaningless (we can trivially work around it) and volatile (e.g. PATH modifications).

`uname` isn't a shell script, it's a posix standard syscall. examining the result of the syscall is easiest through the binary command `uname`, so I chose to conflate the two, but that's not really relevant. If you prefer, I'll alter the PR to call the libc function directly instead of going through all these layers of indirection; but that's just not the unix way :)